### ISSUE TITLE
java.lang.Throwable: Access is allowed from Event Dispatch Thread (EDT) only

### ISSUE THREAD

**Issue by Petrox** [leinardi/pylint-pycharm#103]
### Step 1: Are you in the right place?
- [x] I have verified there are no duplicate active or recent bugs, questions, or requests
- [x] I have verified that I am using the latest version of the plugin.

### Step 2: Describe your environment
 - Plugin version: `0.16.2`
 - PyCharm/IDEA version: `2023.3.2` (had the same message with earlier versions too.
 - Pylint version: `3.0.2` (windows, via pip)

#### Steps to reproduce:

Not trivial to trigger, it pops up usually when try to commit, save, so probably on filechange, sometimes after stopping the running process (the app I'm developing, but that could also trigger a filesystem change).

```Gradle
java.lang.Throwable: Access is allowed from Event Dispatch Thread (EDT) only
	at com.intellij.openapi.vcs.changes.ui.ChangesTree.getInclusionModel(ChangesTree.java:549)
	at com.intellij.openapi.vcs.changes.ui.ChangesTree.getIncludedSet(ChangesTree.java:617)
	at com.intellij.openapi.vcs.changes.ui.VcsTreeModelData$IncludedUnderData.<init>(VcsTreeModelData.java:255)
	at com.intellij.openapi.vcs.changes.ui.VcsTreeModelData.included(VcsTreeModelData.java:54)
	at com.intellij.vcs.commit.ChangesViewCommitPanel.getIncludedChanges(ChangesViewCommitPanel.kt:191)
	at com.intellij.vcs.commit.AbstractCommitWorkflowHandlerKt.getIncludedPaths(AbstractCommitWorkflowHandler.kt:42)
	at com.intellij.vcs.commit.CommitProjectPanelAdapter.getVirtualFiles(CommitProjectPanelAdapter.kt:29)
	at com.leinardi.pycharm.pylint.handlers.ScanFilesBeforeCheckinHandler$2.run(ScanFilesBeforeCheckinHandler.java:114)
	at com.intellij.openapi.progress.impl.CoreProgressManager.startTask(CoreProgressManager.java:434)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.startTask(ProgressManagerImpl.java:132)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcessWithProgressSynchronously$8(CoreProgressManager.java:518)
	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$new$0(ProgressRunner.java:84)
	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$submit$4(ProgressRunner.java:248)
	at com.intellij.openapi.progress.ProgressManager.lambda$runProcess$0(ProgressManager.java:73)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$1(CoreProgressManager.java:192)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$executeProcessUnderProgress$12(CoreProgressManager.java:610)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:685)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computeUnderProgress(CoreProgressManager.java:641)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:609)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:78)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:179)
	at com.intellij.openapi.progress.ProgressManager.runProcess(ProgressManager.java:73)
	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$submit$5(ProgressRunner.java:248)
	at com.intellij.openapi.progress.impl.ProgressRunner$ProgressRunnable.run(ProgressRunner.java:498)
	at com.intellij.util.concurrency.ChildContext$runAsCoroutine$1.invoke(propagation.kt:81)
	at com.intellij.util.concurrency.ChildContext$runAsCoroutine$1.invoke(propagation.kt:81)
	at com.intellij.util.concurrency.ChildContext.runAsCoroutine(propagation.kt:86)
	at com.intellij.util.concurrency.ChildContext.runAsCoroutine(propagation.kt:81)
	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$launchTask$18(ProgressRunner.java:464)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:702)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:699)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1.run(Executors.java:699)
	at java.base/java.lang.Thread.run(Thread.java:840)
```

#### Observed Results:

I can close the dialog from Pycharm and then the editor seems to work, then in 10 minutes I'll get a new popup with the same exception.
  
#### Expected Results:

Same boring cruise, with basic unused variables and line too longs.

#### Relevant Code:

Don't know where to look or which part is the cause, looks like internal ide thread conflict, not something that depends on the code or config.

FYI I have github copilot, jetbrains ai something and jetbrains full line completion enabled at the same time, which is not normal, but it shouldn't matter for the pylint plugin much.

**Comment by leinardi** [comment]
Looking at the stacktrace it seems that the issue is happening inside `com.intellij.openapi`, could it be that this is something that they have to fix?

**Comment by bewczardski** [comment]
It looks like this is a change in the 2023.3 Threading Model by Intellij:

- https://plugins.jetbrains.com/docs/intellij/general-threading-rules.html?from=jetbrains.org#read-write-lock

Found that link from this support post in the IntelliJ Community forums: https://intellij-support.jetbrains.com/hc/en-us/community/posts/7602780460306-Access-is-allowed-from-event-dispatch-thread-only

**Comment by Petrox** [comment]
Great find, looks relevant by the title.

**Comment by leinardi** [comment]
I just made a PR that should fix this issue: #106 
Can someone please test and give a confirmation of the fix?

**Comment by bewczardski** [comment]
tested with IntelliJ IDEA 2024.1.1 (Ultimate Edition), ran without any issues

**Comment by leinardi** [comment]
Thank you! I just made a new release, it should be out soon, just waiting for JetBrains approval :+1:

**Comment by bewczardski** [comment]
Apparently I'm not very good at reading version numbers, I thought I was using the latest version but tested with 0.16.2, I just installed 0.16.3 and I'm getting an error related to another PR: https://github.com/leinardi/pylint-pycharm/issues/105

`com.intellij.diagnostic.PluginException: `ActionUpdateThread.OLD_EDT` is deprecated and going to be removed soon. 'com.leinardi.pycharm.pylint.actions.ScrollToSource' must override `getActionUpdateThread()` and chose EDT or BGT. See ActionUpdateThread javadoc. [Plugin: com.leinardi.pycharm.pylint]
	at com.intellij.diagnostic.PluginProblemReporterImpl.createPluginExceptionByClass(PluginProblemReporterImpl.java:23)
	at com.intellij.diagnostic.PluginException.createByClass(PluginException.java:90)
	at com.intellij.diagnostic.PluginException.reportDeprecatedUsage(PluginException.java:125)
	at com.intellij.openapi.actionSystem.ActionUpdateThreadAware.getActionUpdateThread(ActionUpdateThreadAware.java:21)
	at com.intellij.openapi.actionSystem.AnAction.getActionUpdateThread(AnAction.java:201)
`
![image](https://github.com/user-attachments/assets/d201ba94-e184-423d-8c4d-dd1c7271e89c)

**Comment by leinardi** [comment]
@bewczardski thanks! Yeah I forgot to apply the fix to `com.leinardi.pycharm.pylint.actions.ScrollToSource`. I just released 0.16.4 with a fix for this.

**Comment by bewczardski** [comment]
Tested 0.16.4 and that error did not occur, everything looks good!
